h2/src/main/org/h2/mvstore/tx/TransactionStore.java

```diff
@@ -881,7 +881,7 @@ static String getDataTypeRegistrationKey(DataType<?> dataType) {
                 registerDataType(getKeyType());
             }
             if (getValueType() == null) {
-                setValueType((DataType<? super V>) new VersionedValueType<V,Object>(defaultDataType));
+                setValueType((DataType<? super V>) new VersionedValueType<>(defaultDataType));
```

Comment:
After this change cast also needs to be removed to avoid compiler warning.

Reply:
Sorry but, I have rebuilt H2 with maven and I have not seen the warning.